### PR TITLE
Convert to Go modules

### PR DESCRIPTION
**Pull request by hisaac**
### Checklist
- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

> I'm not sure if converting to modules is a major, minor, or patch change. Please advise.

### Context

While working on https://github.com/bitrise-steplib/steps-xcode-build-for-simulator/pull/31, I ran into an issue with the `go-xcode` dependencies. I tried updating it using `dep`, but was unsuccessful, so I thought I'd try my hand at converting the repo to use Go modules instead, and then update it.

### Changes

- Added the `.idea` directory go the `.gitignore`.
- Converted the repo to use Go modules.
    - This included deleting the `vendor` directory, as I think that's not needed.
- Updated the `go-xcode` dependency to point to the [v1.0.6](https://github.com/bitrise-io/go-xcode/releases/tag/v1.0.6) tag.
    - Updating this dependency was required to work around an issue with parsing the output of the xcodebuild version command (https://github.com/bitrise-io/go-xcode/pull/152).
    - This also necessitated making a small change to the `main.go` file, as the `SetDisableIndexWhileBuilding` method was removed from `build.go` in (https://github.com/bitrise-io/go-xcode/pull/121).

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -241,8 +241,10 @@ func main() {
 			xcodeBuildCmd.SetCustomOptions(options)
 		}
 
-		// Disabe indexing while building
-		xcodeBuildCmd.SetDisableIndexWhileBuilding(cfg.DisableIndexWhileBuilding)
+		// Disable indexing while building
+		if cfg.DisableIndexWhileBuilding {
+			xcodeBuildCmd.SetCustomBuildAction("COMPILER_INDEX_STORE_ENABLE=NO")
```
Comment (shams-ahmed) [inline main.go:246]: this may override line 232.. what about adding a []string that collects all the custom action than apply it later on the command?

**Review by shams-ahmed — CHANGES_REQUESTED** [review]

**Comment by shams-ahmed** [comment]
Also in the Bitrise.yml file please remove `dep-update:`.

And I don't think we will need `Gopkg.toml` and `Gopkg.lock`

File: main.go (hunk 1 above)
Comment (hisaac) [reply main.go]: Good catch! Should be fixed in https://github.com/bitrise-steplib/steps-xcode-build-for-simulator/pull/32/commits/fce02bbd95fb8cfcfffa0d9c28fc9e9b5bde498f

**Comment by hisaac** [comment]
> Also in the Bitrise.yml file please remove `dep-update:`.
> 
> And I don't think we will need `Gopkg.toml` and `Gopkg.lock`

Should be fixed in https://github.com/bitrise-steplib/steps-xcode-build-for-simulator/pull/32/commits/fce02bbd95fb8cfcfffa0d9c28fc9e9b5bde498f 👍

**Review by shams-ahmed — APPROVED** [review]

**Comment by shams-ahmed** [comment]
@hisaac looking good!

**Comment by hisaac** [comment]
I should also update the version number. Would you say this would be a minor version bump? Or patch?

**Comment by shams-ahmed** [comment]
@hisaac let do it on the other PR. merging to master is not classed as a deploy

**Pull request merged**